message.go

```diff
@@ -120,6 +121,17 @@ func NewMessage(data []byte) *Message {
 	}
 }
 
+// NewMessageWithValue returns a *Message with the Value field set.
+// valueSection is set to true so even if the Value field is nil,
+// the AMQPValue section will be written as NullValue
+// that's a simplification of what the .NET AMQP parser does with sections
+func NewMessageWithValue(value any) *Message {
+	return &Message{
+		valueSection: true,
```

Comment:
This will work. However, my concern is that in order to get this behavior you have to use this constructor. i.e. there's no way to replicate the behavior when creating a raw message (`msg := &Message{}`). So, it's a bit "magical".

Another thought I had was to introduce a `NullValue` which one would set when constructing the message.
```go
msg := &amqp.Message{
  Value: amqp.NullValue{},
}
```
I like that it's explicit but doesn't have great discoverability.

@richardpark-msft what are your thoughts on this, maybe you have another idea?

Reply:
Indeed, we'd have to have some special handling for `NullValue` to encode it as `null`.

Reply:
I whipped up https://github.com/Azure/go-amqp/pull/336 with what it would look like.